plugins/imfile/imfile.c

```diff
@@ -702,27 +714,24 @@ detect_updates(fs_edge_t *const edge)
 {
 	act_obj_t *act;
 	struct stat fileInfo;
+	int restart = 0;
 
-	for(act = edge->active ; act != NULL ; ) {
+	for(act = edge->active ; act != NULL ; act = act->next) {
 		DBGPRINTF("detect_updates checking active obj '%s'\n", act->name);
-		const int r = stat(act->name, &fileInfo);
+		const int r = lstat(act->name, &fileInfo);
 		if(r == -1) { /* object gone away? */
 			DBGPRINTF("object gone away, unlinking: '%s'\n", act->name);
-			act_obj_t *toDel = act;
-			act = act->next;
-			DBGPRINTF("new next act %p\n", act);
-			act_obj_unlink(toDel);
-			continue;
+			act_obj_unlink(act);
+			restart = 1;
+			break;
```

Comment:
I had to change a flow now because unlink call can do more thinks due to additional target removals, so if I unlink anything flom list I abort completely and go trough whole function again so I have ensured coherent state of the list.

Reply:
ok, let me first try to update to clang-6.0 analyzer. Maybe it solves the situation. This also sounds Iike I do need to go more in-depth over the PR. I'll ping you when clang-6 is ready, not sure I will mange this today.